var/spack/repos/builtin/packages/protobuf/package.py

```diff
@@ -46,6 +47,17 @@ class Protobuf(CMakePackage):
     # first fixed in 3.4.0: https://github.com/google/protobuf/pull/3406
     patch('pkgconfig.patch', when='@:3.3.2')
 
+    def fetch_remote_versions(self):
+        """fix for https://github.com/LLNL/spack/issues/5356"""
+        return dict(map(lambda u:
+                        (u, self.url_for_version(Version(u))),
+                        self.versions))
+
+    def url_for_version(self, version):
```

Comment:
Is this strictly required if the `fetch_remote_versions` function is overridden? The definition in `Package` appears to do a direct replacement vs. searching for a `releases/` directory.

Reply:
since I do need that functionality in the member function above, and it's exactly what `url_for_version` would do I think it's useful to keep it in a separate call, no?